lib/verify-jwt.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+const jwt = require('jsonwebtoken')
+const JwksClient = require('jwks-rsa')
+module.exports = async (token /* : string */, iss /* : string */) => {
+  try {
+    const { header } = jwt.decode(token, { complete: true })
+    const publicKey = await getPublicKey(iss, header.kid)
+    const decoded = jwt.verify(token, publicKey, {
+      algorithms: ['RS256'],
+      issuer: iss,
+    })
+    return decoded
+  } catch (err) {
+    throw new Error('The supplied JWT was invalid.')
+  }
+}
+
+const getPublicKey = (
+  iss /* : string */,
+  kid /* : string */,
+) /* : Promise<string> */ => {
+  const jwksClient = new JwksClient({
+    cache: true,
+    jwksUri: `${iss}/.well-known/jwks.json`,
+  })
```

Comment:
This needs to be created outside this function to allow for caching.

Reply:
Do you mean in the function calling it?

Reply:
I mean it needs to move to line 17. Can can discuss in person if you'd like.

Reply:
It needs the iss property though?